client/cli/src/lib.rs

```diff
@@ -229,78 +227,72 @@ pub trait SubstrateCli: Sized {
+		// Set warn logging by default for some modules.
+		.add_directive("cranelife_wasm=warn".parse().expect("provided directive is valid"))
+		.add_directive("hyper=warn".parse().expect("provided directive is valid"))
+		// Always log the special target `sc_tracing`, overrides global level.
+		.add_directive("sc_tracing=trace".parse().expect("provided directive is valid"))
+		// Enable info for others.
+		.add_directive(tracing_subscriber::filter::LevelFilter::INFO.into());
 
 	if let Ok(lvl) = std::env::var("RUST_LOG") {
-		builder.parse_filters(&lvl);
+		if lvl != "" {
+			// We're not sure if log or tracing is available at this moment, so silently ignore the
+			// parse error.
+			if let Ok(directive) = lvl.parse() {
+				env_filter = env_filter.add_directive(directive);
+			}
+		}
+	}
+
+	if pattern != "" {
+		// We're not sure if log or tracing is available at this moment, so silently ignore the
+		// parse error.
+		if let Ok(directive) = pattern.parse() {
+			env_filter = env_filter.add_directive(directive);
+		}
 	}
 
-	builder.parse_filters(pattern);
 	let isatty = atty::is(atty::Stream::Stderr);
 	let enable_color = isatty;
 
-	builder.format(move |buf, record| {
-		let now = time::now();
-		let timestamp =
-			time::strftime("%Y-%m-%d %H:%M:%S", &now).expect("Error formatting log timestamp");
-
-		let mut output = if log::max_level() <= log::LevelFilter::Info {
-			format!(
-				"{} {}",
-				Colour::Black.bold().paint(timestamp),
-				record.args(),
-			)
-		} else {
-			let name = ::std::thread::current()
-				.name()
-				.map_or_else(Default::default, |x| {
-					format!("{}", Colour::Blue.bold().paint(x))
-				});
-			let millis = (now.tm_nsec as f32 / 1000000.0).floor() as usize;
-			let timestamp = format!("{}.{:03}", timestamp, millis);
-			format!(
-				"{} {} {} {}  {}",
-				Colour::Black.bold().paint(timestamp),
-				name,
-				record.level(),
-				record.target(),
-				record.args()
-			)
-		};
-
-		if !isatty && record.level() <= log::Level::Info && atty::is(atty::Stream::Stdout) {
-			// duplicate INFO/WARN output to console
-			println!("{}", output);
+	let subscriber = tracing_subscriber::FmtSubscriber::builder()
+		.with_env_filter(env_filter)
+		.with_target(false)
+		.with_ansi(enable_color)
+		.with_writer(std::io::stderr)
+		.compact()
+		.finish();
+
+	if let Some(tracing_targets) = tracing_targets {
+		let profiling = sc_tracing::ProfilingLayer::new(tracing_receiver, &tracing_targets);
+
+		if tracing::subscriber::set_global_default(subscriber.with(profiling)).is_err() {
+			tracing::debug!(
+				"💬 Not registering Substrate subscriber, as there is already a global subscriber registered!"
+			);
 		}
-
-		if !enable_color {
-			output = kill_color(output.as_ref());
+	} else {
+		if tracing::subscriber::set_global_default(subscriber).is_err() {
+			tracing::warn!(
+				"💬 Not registering Substrate subscriber, as there is already a global subscriber registered!"
+			);
```

Comment:
Previously it was an `info!` log, now it's either `debug!` or `warn!`. If this is intended, could add some comments explaining why.

Reply:
@bkchr mentioned that it will spam parachains otherwise. https://github.com/paritytech/substrate/pull/6825#pullrequestreview-485791529

Reply:
Yeah, we spawn two nodes in Cumulus and as we mostly use the default "way" of initializing the nodes we will always end in a situation where the logger is already registered.

Reply:
looking at it again, I wonder: why not escalate the error? Here and from `init`(which is already `Result`). And for cumulus I wonder whether we should then refactor the code to be usable without calling `init` (which states in its docs that it "must be done only once."). Which from a brief look at it, shouldn't be too hard... like, if `init` was called once (it doesn't even really need the `&self`) there is no need to call it again... and in cumulus cases we would then only call it once...